File: iFood/FoodTableViewCell.m
//
//  MenuTableViewCell.m
//  iFood
//
//  Created by Ezequiel Gutierrez on 15/9/16.
//  Copyright © 2016 Acamica. All rights reserved.
//

#import "FoodTableViewCell.h"

@interface FoodTableViewCell ()

@property (nonatomic, weak) IBOutlet UILabel *titleLabel;
@property (nonatomic, weak) IBOutlet UILabel *descriptionLabel;
@property (nonatomic, weak) IBOutlet UIImageView *foodImageView;
@property (nonatomic, weak) IBOutlet UIView *containerView;

@end

@implementation FoodTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    
    self.containerView.layer.cornerRadius = 10.0;
    self.containerView.layer.masksToBounds = YES;
    
    self.foodImageView.layer.cornerRadius = 4.0;
    self.foodImageView.layer.masksToBounds = YES;

}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (void) setTile:(NSString*)title description:(NSString*)description andImageName:(NSString*)imageName{
    [self.titleLabel setText:title];
    [self.descriptionLabel setText:description];
    [self.foodImageView setImage:[UIImage imageNamed:imageName]];
}

@end

File: iFood/Views/FoodTableViewCell.swift
//
//  FoodTableViewCell.swift
//  iFood
//
//  Created by Matias Fernandez on 10/05/2017.
//  Copyright © 2017 Acamica. All rights reserved.
//

import UIKit

class FoodTableViewCell: UITableViewCell {

    @IBOutlet private weak var titleLabel:UILabel!
    @IBOutlet private weak var descriptionLabel:UILabel!
    @IBOutlet private weak var foodImageView:UIImageView!
    @IBOutlet private weak var containerView: UIView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

    func updateData(_ food: Food) {
        titleLabel?.text = food.name
        descriptionLabel?.text = food.foodDescription
        foodImageView?.image = UIImage(named: food.foodImage)
        foodImageView.contentMode = .scaleAspectFit
    }
    
}
